### DISCUSSION TITLE
Why can I pull ssh key without password, in terminal, but not through script?

### DISCUSSION THREAD

**Discussion by cesarmsj** [community/community#22396] [New to GitHub]
<p>It’s a server running Ubuntu 20.04. With the root user, in the terminal, if I run git pull, it works ok, but if nginx runs a script for automatic deployment, it doesn’t do git pull, the message below appears:</p>
<p><strong>Host key verification failed.</strong><br>
<strong>fatal: Could not read from remote repository.</strong></p>
<p><strong>Please make sure you have the correct access rights</strong><br>
<strong>and the repository exists.</strong></p>
<p>See the ssh configuration file, in ~/.ssh/config:</p>
<p>Host <a href="http://github.com" rel="noopener nofollow ugc">github.com</a><br>
Hostname <a href="http://github.com" rel="noopener nofollow ugc">github.com</a><br>
IdentityFile /var/www/deploy/.ssh/id_ed25519<br>
IdentitiesOnly yes</p>
<p>Key Permissions:</p>
<p>-rwxrwxrwx 1 _nginx _nodejs  411 Nov 24 14:52 id_ed25519</p>
<p>The goal is: a webhook calls for a url that calls a shell script that auto-deploys it on the server.</p>

**Comment by airtower-luna** [comment, marked as answer]
<aside class="quote no-group" data-username="cesarmsj" data-post="1" data-topic="214256">
<div class="title">
<div class="quote-controls"></div>
 cesarmsj:</div>
<blockquote>
<p><strong>Host key verification failed.</strong><br>
<strong>fatal: Could not read from remote repository.</strong></p>
</blockquote>
</aside>
<p>The error message is about the <em>host key</em>, not your identity key. The host key is what GitHub’s SSH server uses for authentication, to ensure you’re talking to the right server (and not, say, deploying code provided by some MITM attacker).</p>
<p>The correct host key must be listed in the user’s <code>~/.ssh/known_hosts</code> file. You could either copy the relevant line from yours, or connect interactively as the nginx user and confirm after checking the fingerprint against this list: <a href="https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/githubs-ssh-key-fingerprints" class="inline-onebox">GitHub's SSH key fingerprints - GitHub Docs</a></p>
<aside class="quote no-group" data-username="cesarmsj" data-post="1" data-topic="214256">
<div class="title">
<div class="quote-controls"></div>
 cesarmsj:</div>
<blockquote>
<p>With the root user, in the terminal, if I run git pull, it works ok</p>
</blockquote>
</aside>
<p>As a side note, please avoid running things as root that don’t actually need root access. It’s an unnecessary risk. :slightly_smiling_face:</p>

**Comment by cesarmsj** [comment]
<p>Thank you very much for the answer. Unfortunately, it didn’t work out. Includes the following line:</p>
<p><a href="http://github.com" rel="noopener nofollow ugc">github.com</a> ssh-ed25519 SHA256:+DiY3wvvV6TuJJhbpZisF/zLDA0zPMSvHdkr4UvCOqU</p>
<p>in /var/www/.ssh/know_hosts, restarted ssh, and the message still continues to appear.</p>
<p>I left /var/www/.ssh because looking in /etc/passwd, there it is:</p>
<p>_nginx:x:998:1002::/var/www:/bin/false</p>

**Comment by cesarmsj** [comment]
<p>Oops, I solved the case haha. The correct user is _nodejs and not _nginx. So your answer helped me, thank you very much!</p>

**Comment by airtower-luna** [comment]
<aside class="quote no-group" data-username="cesarmsj" data-post="3" data-topic="214256">
<div class="title">
<div class="quote-controls"></div>
 cesarmsj:</div>
<blockquote>
<pre><code class="lang-auto">github.com ssh-ed25519 SHA256:+DiY3wvvV6TuJJhbpZisF/zLDA0zPMSvHdkr4UvCOqU
</code></pre>
</blockquote>
</aside>
<p>You can’t copy the fingerprint for the last part, check your own <code>.ssh/known_hosts</code> file to see the correct format. This should work:</p>
<pre><code class="lang-auto">github.com ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAIOMqqnkVzrm0SdG6UOoqKLsabgH5C9okWi0dh2l9GKJl
</code></pre>
<p>You can get the fingerprint from that with <code>ssh-keygen -l</code> to verify.</p>
